malloc.c

```diff
@@ -611,7 +611,7 @@ GC_API void GC_CALL GC_free(void * p)
                 /* object is reallocated, it doesn't matter.  O.w. the  */
                 /* collector will do it, since it's on a free list.     */
         if (ok -> ok_init && EXPECT(sz > sizeof(word), TRUE)) {
-            BZERO((word *)p + 1, sz-sizeof(word));
+            BZERO((void **)p + 1, sz-sizeof(word *));
```

Comment:
Should this be `word **` or `sizeof(void *)` -- basically I was expecting to see the type symmetry be maintained in case that's important.

Also, I wondered if this was supposed to be `if defined(__CHERI_PURE_CAPABILITY__)` gated?

Reply:
I agree with you on both counts @ltratt.   Fixed in 
[c8fa349](https://github.com/capablevms/bdwgc/pull/20/commits/c8fa349dba6d7cdad90bf09081a7fb81f358165d)